drivers/pds/pdsbackup/backuputils.go

```diff
@@ -0,0 +1,242 @@
+	accountName := backupClient.azureStorageClient.accountName
+	backupCred, err := backupClient.components.BackupCredential.CreateAzureBackupCredential(tenantId, name, accountKey, accountName)
+	if err != nil {
+
+		return nil, fmt.Errorf("Error in adding the backup credentials to PDS , Err: %v ", err)
+	}
+	log.Infof("Adding backup target {Name: %v} to PDS.", name)
+	backupTarget, err := backupClient.components.BackupTarget.CreateBackupTarget(tenantId, name, backupCred.GetId(), bucketName, "", "azure")
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create Azure backup target, Err: %v ", err)
+	}
+	log.Infof("[Backup Target: %v]Syncing to target clusters", name)
+	return backupTarget, nil
+}
+
+// CreateGcpBackupCredsAndTarget create backup creds,bucket and target.
+func (backupClient *BackupClient) CreateGcpBackupCredsAndTarget(tenantId, name string) (*pds.ModelsBackupTarget, error) {
+	log.Info("Create google storage for adding GCP backup target to control plane.")
+	err := backupClient.gcpStorageClient.createBucket(bucketName)
+	if err != nil {
+		return nil, fmt.Errorf("Failed while creating bucket, Err: %v ", err)
+	}
+	log.Info("Add GCP backup credentials")
+	projectId := backupClient.gcpStorageClient.projectId
+	key := "GCP_JSON_PATH"
+	gcpJsonKey, isExist := os.LookupEnv(key)
+	if !isExist {
+		return nil, fmt.Errorf("environment var: %v doesn't exist", key)
+	}
+	backupCred, err := backupClient.components.BackupCredential.CreateGoogleCredential(tenantId, name, projectId, gcpJsonKey)
+	if err != nil {
+		return nil, fmt.Errorf("Error in adding the backup credentials to PDS , Err: %v ", err)
+	}
+	log.Infof("Adding backup target {Name: %v} to PDS.", name)
+	backupTarget, err := backupClient.components.BackupTarget.CreateBackupTarget(tenantId, name, backupCred.GetId(), bucketName, "", "google")
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create google backup target, Err: %v ", err)
+	}
+	return backupTarget, nil
+
+}
+
+// DeleteAwsS3BackupCredsAndTarget delete backup creds,bucket and target.
+func (backupClient *BackupClient) DeleteAwsS3BackupCredsAndTarget(backupTargetId string) error {
+	log.Info("Removing S3 backup creadentials and target from PDS.")
+	backupTarget, _ := backupClient.components.BackupTarget.GetBackupTarget(backupTargetId)
+	credId := backupTarget.GetBackupCredentialsId()
+	log.Infof("Deleting backup target {Name: %v} to PDS.", backupTarget.GetName())
+	_, err := backupClient.components.BackupTarget.DeleteBackupTarget(backupTargetId)
+	if err != nil {
+		return fmt.Errorf("Failed to delete AWS S3 backup target, Err: %v ", err)
+	}
+	waitErr := wait.Poll(bkpTimeInterval, bkpMaxtimeInterval, func() (bool, error) {
+		model, bkpErr := backupClient.components.BackupTarget.GetBackupTarget(backupTargetId)
+		if model != nil {
+			log.Info(model.GetName())
+			return false, bkpErr
+		}
+		if bkpErr != nil && strings.Contains(bkpErr.Error(), "not found") {
+			return true, nil
```

Comment:
if "not found" is expected shouldn't it return false

Reply:
not found is expected, hence returning true.
Otherwise, it's returning a false, Error.